pkg/ottl/ottlfuncs/func_timestamp.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TimestampArguments[K any] struct {
+	Time   ottl.TimeGetter[K]
+	Format string
+}
+
+func NewTimestampFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Timestamp", &TimestampArguments[K]{}, createTimestampFunction[K])
```

Comment:
Should we use `FormatTime` instead? The word `Timestamp` can denote an integer representation of a time.